corehq/apps/smsbillables/models.py

```diff
@@ -407,25 +408,23 @@ def _get_twilio_client():
             config = twilio_backend.config
             return TwilioRestClient(config.account_sid, config.auth_token)
 
-        try:
-            if backend_message_id:
+        if backend_message_id:
+            try:
                 twilio_message = _get_twilio_client().messages.get(backend_message_id)
-                return _TwilioChargeInfo(
-                    Decimal(twilio_message.price) * -1,
-                    SmsGatewayFee.get_by_criteria(
-                        SQLTwilioBackend.get_api_id(),
-                        direction,
-                    )
-                )
-            else:
-                log_smsbillables_error(
-                    "Could not create gateway fee for Twilio message %s: no backend_message_id" % couch_id
+            except TwilioRestException:
+                raise RetryBillableTaskException
```

Comment:
we should also check the status of the message and if it's accepted, queued, sending, or receiving then we should also raise the same retry exception

https://www.twilio.com/docs/api/rest/message#sms-status-values